Sources/SpelledPitch/PitchSpeller/Wetherfield/SpellingInverter.swift

```diff
@@ -100,6 +109,44 @@ extension SpellingInverter {
         return pitchedDependencies.reduce(into: [:], dependeciesReducer)
     }
     
+    func generateWeightsFromCycles (_ dependencies: [PitchedEdge: Set<PitchedEdge>])
+        -> [PitchedEdge: Double] {
+        let clumper = findStronglyConnectedComponent(dependencies)
+        let directedAcyclicGraph = groupNodes(dependencies, clumper)
```

Comment:
Not suggesting we do this, but would it ever be feasible to implement a DAG at the type level?

Reply:
Interesting. The only way I could think of doing this would be storing paths (as a kind of "ordered set"), which would probably be quite space inefficient, but it would be nice if this were possible in a way that was not too costly